### ISSUE TITLE
Div by 0 in fluid_player_set_tempo

### ISSUE THREAD

**Issue by bunnylin** [FluidSynth/fluidsynth#1111]
### FluidSynth version
2.2.7 on Archlinux

### Describe the bug
If you call `fluid_player_set_tempo(player, FLUID_PLAYER_TEMPO_EXTERNAL_MIDI, midiTempo)` after loading a file but before starting playback (or immediately after starting), FluidSynth crashes with div by zero. Because the player hasn't yet parsed the given file's header, player->division is still 0.

```
Thread 1 "test" received signal SIGFPE, Arithmetic exception.
0x00007ffff7e87b3f in fluid_player_update_tempo (player=0x8b3100) at /usr/src/debug/fluidsynth-2.2.7/src/midi/fluid_midi.c:2347
2347            deltatime = (float)tempo / (float)player->division / (float)1000.0;
```

### Expected behavior
Either return FLUID_FAILED from `fluid_player_set_tempo`, or stash the given tempo and apply it after the file header is parsed and division is known. The function's documentation should perhaps have a sentence about this.

**Comment by jjceresa** [comment]
Could be fixed in fluid_player_update_tempo() that do nothing if player->division is 0. This way the given tempo set by fluid_player_set_tempo() is always taken into account and applied only as soon the file header is parsed and the division is known.

static void fluid_player_update_tempo(fluid_player_t *player)
{
    int tempo; /* tempo in micro seconds by quarter note */
    float deltatime;

    /* do nothing until the division is know */
    if(player->division == 0)
    {
        return;
    }

   /* update internal deltatime dependant of actual tempo */
    .
    .
}

**Comment by derselbst** [comment]
@bunnylin Thanks for the report.
@jjceresa I agree with your proposal and created a fix in #1116
